src/pudl/settings.py

```diff
@@ -536,6 +536,8 @@ def check_eia_dependencies(cls, data: dict[str, Any]) -> dict[str, Any]:
 class DatasetsSettings(FrozenBaseModel):
     """An immutable pydantic model to validate PUDL Dataset settings."""
 
+    etl_type: Literal["fast", "full", "adhoc"] = "adhoc"
```

Comment:
@zaneselvans we talked about adding some way of identifying what 'type' of job we're running. here it is.